agent/shadow_helper.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+#
+# This script is a helper to generate shadow file.
+# shellhub agent support to define shadow path.
+#
+# example:
+#
+# [root mode] (default)
+#  Only can define users that are in system.
+#  The shadow file defines what accounts can access ssh via shellhub.
+#  The passwords of system and shellhub could be different.
+#
+# [user mode]
+#  it allows different accounts to login but same system account.
+#  There is no limit to define connection username.
+#  You can review the session log in web console.
+#
+# ```
+#   [user@node agent]$ USER=test PASSWD=123 SHADOW_FILEANAME=.shadow ./shadow_helper.sh
+#   [user@node agent]$ SHADOW_FILEANAME=.shadow ./agent
+#
+#   $ ssh test@sshid
+#   test@sshid's password:     ### << type password "123" to login
+#   [user@node ~]$             ### << login as user who run ./agent
+# ```
+
+##
+## USER
+## PASSWD
+## SHADOW_FILEANAME
+##
+
+if [ -z ${USER+x} ]; then
+   echo "USER is unset"
+   exit 1
+fi
+
+if [ -z ${PASSWD+x} ]; then
+   echo "PASSWD is unset"
+   exit 1
+fi
+
+if [ -z ${SHADOW_FILEANAME+x} ]; then
+   echo "SHADOW_FILEANAME is unset"
+   exit 1
+fi
+
+ENCRYPTED_PASSWD=$(echo "$PASSWD" | mkpasswd --method=SHA-512 --stdin)
+echo $USER:$ENCRYPTED_PASSWD::::::: >> ${SHADOW_FILEANAME}
```

Comment:
The shadow file generation could be done using a sub-command inside `shellhub-agent` itself.  This allows the deployment of a single binary. 

For example:

```
./agent create-shadow-file <username> -o <destination file>
```

and it could read the password from stdin, for example. @gustavosbarreto what do you think?

Reply:
It is a good idea.

If https://github.com/shellhub-io/shellhub/pull/597 be merged, that would be easy to implement.

Reply:
@otavio @gustavosbarreto it is good to have sub-commands inside shellhub-agent.

Do you think let put commands/sub-commands related code into pkg/cmd/<sub-commands>.go?
Or even consider to import cobra.